diff -uNr libstatgrab-orig/configure.in libstatgrab/configure.in
--- libstatgrab-orig/configure.in	2004-05-30 21:56:28.000000000 +0200
+++ libstatgrab/configure.in	2004-06-09 19:00:00.000000000 +0200
@@ -40,6 +40,12 @@
                 [],
                 [#include <sys/disk.h>])
 
+# ds_bytes has been removed in OpenBSD
+AC_CHECK_MEMBER(struct disk_sysctl.ds_bytes,
+                [AC_DEFINE(HAVE_DS_BYTES, , [Removed in OpenBSD])],
+                [],
+                [#include <sys/disk.h>])
+
 # Check for sys/loadavg.h
 AC_CHECK_HEADERS([sys/loadavg.h])
 
diff -uNr libstatgrab-orig/src/libstatgrab/disk_stats.c libstatgrab/src/libstatgrab/disk_stats.c
--- libstatgrab-orig/src/libstatgrab/disk_stats.c	2004-05-30 21:56:28.000000000 +0200
+++ libstatgrab/src/libstatgrab/disk_stats.c	2004-06-09 19:01:26.000000000 +0200
@@ -415,6 +415,12 @@
 #endif
 #else
 		rbytes = wbytes = stats[i].ds_bytes;
+#ifdef HAVE_DS_BYTES
+                rbytes = wbytes = stats[i].ds_bytes;
+#else
+                rbytes = stats[i].ds_rbytes;
+                wbytes = stats[i].ds_wbytes;
+#endif
 #endif
 
 		/* Don't keep stats for disks that have never been used. */
